src/gui/guiChatConsole.cpp

```diff
@@ -328,6 +348,15 @@ void GUIChatConsole::drawText()
 			core::rect<s32> destrect(
 				x, y, x + m_fontsize.X * fragment.text.size(), y + m_fontsize.Y);
 
+
+			core::recti rect;
+			if (m_scrollbar_enabled && m_scrollbar->isVisible())
+				// leave 4 pixels of space between scrollbar and text
+				rect = core::rect<s32> (0, 0, m_screensize.X - 32 - 4, m_height);
```

Comment:
needs to take display density and perhaps gui_scaling into account (also deduplicate it)

Reply:
This is the only time i construct this rectangle in the code. However, now it also uses getScrollbarSIze() instead of hardcoded values.